pkg/apis/serving/v1alpha1/local_model_node_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type SourceDestination struct {
+	SourceURI   string `json:"sourceUri"`
+	Destination string `json:"destination"`
+}
+
+// +k8s:openapi-gen=true
+type LocalModelNodeSpec struct {
+	SourceDestinations []SourceDestination `json:"sourceDestinations"`
+	ModelStatus        *ModelStatus        `json:"modelStatus,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Cluster"
+type LocalModelNode struct {
```

Comment:
Refer to design of Kubernetes node
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#node-v1-core

Reply:
moved model status to LocalModelNodeStatus